src/DateField/DateField.tsx

```diff
@@ -95,13 +123,13 @@ export default class DateField extends React.PureComponent {
 
     this.setState({ isValid, ...patch, isDirty: true }, () => {
       if (isValid && every([this.state.day, this.state.month, this.state.year], s => s.length > 0)) {
-        this.props.onChange(new Date(this.state.year, this.state.month - 1, this.state.day));
+        this.props.onChange(new Date(parseInt(this.state.year), parseInt(this.state.month) - 1, parseInt(this.state.day)));
```

Comment:
hey hey you're changing the logic here!
what happens if `this.state.year` is `"5s"`?